### ISSUE TITLE
Return Nx tensors as a result of predictions

### ISSUE THREAD

**Issue by jeantux** [jeantux/scitree#2]
Return the results of the predictions in the format of Nx tensors.
This idea seems cool considering that other tools have a similar approach, as is the example of `tensorflow_decision_forests` in google collab.

![image](https://user-images.githubusercontent.com/30236552/156248841-99ef0439-2276-42a3-a9d1-a42a8969f1fd.png)

Using Scritree the result is:
![image](https://user-images.githubusercontent.com/30236552/156249229-4ca2e2d3-96bf-4b69-91b8-6234e2dd3a1f.png)

WDYT @polvalente

**Comment by polvalente** [comment]
@jeantux sounds good, altough I'm still not sure if we should return an Explorer Dataframe instead of an Nx Tensor.
Looking at the other tools, what is the input-output relationship? Is it dataframe->array or array->array?

To me, it seems that returning a Tensor would make it easier to feed the model directly into other Nx algorithms, while returning the Dataframe would mostly have the advantage of making each column have an explicit semantic.

Any thoughts?

**Comment by jeantux** [comment]
Normally the relation is dataframe -> array.

A result in dataframe would only be useful for classification, and yet in many cases of values are converted to index/numbers then we would have a data with the following structure (doesn't sound good to me):

```elixir
%{
  "1" => [0.0, 0.0, 0.99], 
  "2" => [0.99, 0.0, 0.0]
}
```

I thought of adding a prediction function with provability (`predict_proba`) and another one with direct function with predictions (`predict`) (both returning tensors), like for example sklearn

```python
>>> from sklearn.datasets import load_iris
>>> from sklearn.linear_model import LogisticRegression
>>> X, y = load_iris(return_X_y=True)
>>> clf = LogisticRegression(random_state=0).fit(X, y)
>>> clf.predict(X[:2, :])
array([0, 0])
>>> clf.predict_proba(X[:2, :])
array([[9.8...e-01, 1.8...e-02, 1.4...e-08],
       [9.7...e-01, 2.8...e-02, ...e-08]])
>>> clf.score(X, y)
0.97...
```
and provide a functions to convert from probability to results 🤔

**Comment by polvalente** [comment]
> Normally the relation is dataframe -> array.
> 
> A result in dataframe would only be useful for classification, and yet in many cases of values are converted to index/numbers then we would have a data with the following structure (doesn't sound good to me):
> 
> ```elixir
> %{
>   "1" => [0.0, 0.0, 0.99], 
>   "2" => [0.99, 0.0, 0.0]
> }
> ```
> 
> I thought of adding a prediction function with provability (`predict_proba`) and another one with direct function with predictions (`predict`) (both returning tensors), like for example sklearn
> 
> ```python
> >>> from sklearn.datasets import load_iris
> >>> from sklearn.linear_model import LogisticRegression
> >>> X, y = load_iris(return_X_y=True)
> >>> clf = LogisticRegression(random_state=0).fit(X, y)
> >>> clf.predict(X[:2, :])
> array([0, 0])
> >>> clf.predict_proba(X[:2, :])
> array([[9.8...e-01, 1.8...e-02, 1.4...e-08],
>        [9.7...e-01, 2.8...e-02, ...e-08]])
> >>> clf.score(X, y)
> 0.97...
> ```
> and provide a functions to convert from probability to results 🤔 
> 

I would go with returning Tensors then!
